src/StarkEx.Crypto.SDK/Hashing/SpotTradingMessageHasher.cs

```diff
@@ -88,21 +111,79 @@ public BigInteger EncodeTransferWithFees(EncodeTransferWithFeesModel encodeTrans
         return pedersenHash.CreateHash(thirdInnerHash, fifthWeight);
     }
 
+    /// <inheritdoc />
+    public BigInteger EncodeTransferWithoutFees(EncodeTransferWithoutFeesModel encodeTransferWithoutFeesModel)
+    {
+        Guards.NotNullOrEmptyOrWhitespace(encodeTransferWithoutFeesModel.AssetIdSold);
+        Guards.NotInvalidHex(encodeTransferWithoutFeesModel.AssetIdSold, nameof(encodeTransferWithoutFeesModel.AssetIdSold));
+        Guards.NotNullOrEmptyOrWhitespace(encodeTransferWithoutFeesModel.ReceiverStarkKey);
+        Guards.NotInvalidHex(encodeTransferWithoutFeesModel.ReceiverStarkKey, nameof(encodeTransferWithoutFeesModel.ReceiverStarkKey));
+
+        var packedMessage = BigInteger.One
+            .ShiftLeft(31).Add(encodeTransferWithoutFeesModel.VaultIdFromSender)
+            .ShiftLeft(31).Add(encodeTransferWithoutFeesModel.VaultIdFromReceiver)
+            .ShiftLeft(63).Add(encodeTransferWithoutFeesModel.QuantizedAmountToTransfer)
+            .ShiftLeft(63).Add(BigInteger.Zero)
+            .ShiftLeft(31).Add(new BigInteger(encodeTransferWithoutFeesModel.Nonce.ToString()))
+            .ShiftLeft(22).Add(new BigInteger((encodeTransferWithoutFeesModel.ExpirationTimestamp / 3600).ToString()));
+
+        var firstInnerHash = pedersenHash.CreateHash(
+            new BigInteger(encodeTransferWithoutFeesModel.AssetIdSold.RemoveHexPrefix(), 16),
+            new BigInteger(encodeTransferWithoutFeesModel.ReceiverStarkKey.RemoveHexPrefix(), 16));
+
+        return pedersenHash.CreateHash(firstInnerHash, packedMessage);
+    }
+
     /// <inheritdoc />
     public BigInteger EncodeConditionalTransferWithFees(EncodeTransferWithFeesModel encodeTransferWithFeesModel)
     {
+        Guards.NotNullOrEmptyOrWhitespace(encodeTransferWithFeesModel.AssetIdSold);
```

Comment:
Shouldn't we validate this at creation of the model object (ie. `EncodeTransferWithFeesModel`), following a similar approach to a regular value object? I don't see a reason why we would ever initialise this model in an invalid state and moving the validation would make this code less verbose.

If decide on moving forward with this we should do it on a separate PR.